editor/store/selectors.js

```diff
@@ -275,7 +276,7 @@ export function isCurrentPostPublished( state ) {
 	const post = getCurrentPost( state );
 
 	return [ 'publish', 'private' ].indexOf( post.status ) !== -1 ||
-		( post.status === 'future' && moment( post.date ).isBefore( moment() ) );
+		( post.status === 'future' && moment.tz( post.date, 'WP' ).isBefore( moment.tz( 'WP' ) ) );
```

Comment:
instead of repeating the same timezone initalization everywhere, is it possible to provide a specific moment constructor in `data/index.js` and always to `import { moment } from '@wordpress/date'` ?

Reply:
yeh, I was thinking the some thing when I was finishing up this change.
It'd be much easier to do it all in one spot.